datarepo-clienttests/src/main/java/disruptivescripts/DeleteInitialPods.java

```diff
@@ -0,0 +1,58 @@
+package disruptivescripts;
+
+import static utils.KubernetesClientUtils.componentLabel;
+
+import io.kubernetes.client.openapi.models.V1Deployment;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import runner.DisruptiveScript;
+import runner.config.TestUserSpecification;
+import utils.KubernetesClientUtils;
+
+public class DeleteInitialPods extends DisruptiveScript {
+  public DeleteInitialPods() {
+    super();
+    manipulatesKubernetes = true;
+  }
+
+  private static final Logger logger = LoggerFactory.getLogger(DeleteInitialPods.class);
+
+  public void disrupt(List<TestUserSpecification> testUsers) throws Exception {
+    logger.info(
+        "Starting disruption - all initially created api pods will be deleted, one by one.");
+
+    V1Deployment apiDeployment = KubernetesClientUtils.getApiDeployment();
+    if (apiDeployment == null) {
+      throw new RuntimeException("API deployment not found.");
+    }
+
+    // get list of api pod names
+    String deploymentComponentLabel = apiDeployment.getMetadata().getLabels().get(componentLabel);
+    List<String> podsToDelete = new ArrayList<>();
+    KubernetesClientUtils.listPods().stream()
+        .filter(
+            pod ->
+                deploymentComponentLabel.equals(pod.getMetadata().getLabels().get(componentLabel)))
+        .forEach(p -> podsToDelete.add(p.getMetadata().getName()));
+
+    // give task time to get started before deleting pods
+    TimeUnit.SECONDS.sleep(15);
```

Comment:
Do you think it would help if we moved the disrupt thread pool kickoff to after the user journey kickoff? Just wondering if all disrupt scripts will want to do this sleep and we could address it for all of them.

Reply:
Good idea - I think that will be common for the disrupt scripts. I moved it to just before we call disrupt. Thoughts?